lib/cli/src/commands/app/secrets/create.rs

```diff
@@ -0,0 +1,271 @@
+use super::utils::Secret;
+use crate::{
+    commands::{
+        app::util::{get_app_config_from_dir, prompt_app_ident, AppIdent, AppIdentFlag},
+        AsyncCliCommand,
+    },
+    opts::{ApiOpts, WasmerEnv},
+};
+use anyhow::Context;
+use colored::Colorize;
+use dialoguer::theme::ColorfulTheme;
+use is_terminal::IsTerminal;
+use std::{
+    collections::{HashMap, HashSet},
+    env::current_dir,
+    path::{Path, PathBuf},
+    str::FromStr,
+};
+use wasmer_api::WasmerClient;
+
+/// Create a new secret related to an Edge app.
+#[derive(clap::Parser, Debug)]
+pub struct CmdAppSecretsCreate {
+    /// The path to the directory where the config file for the application will be written to.
+    #[clap(long = "app-dir", conflicts_with = "app")]
+    pub app_dir_path: Option<PathBuf>,
+
+    /// The name of the secret to create.
+    #[clap(name = "name")]
+    pub secret_name: Option<String>,
+
+    /// The value of the secret to create.
+    #[clap(name = "value")]
+    pub secret_value: Option<String>,
+
+    #[clap(flatten)]
+    pub app_id: AppIdentFlag,
+
+    /// Path to a file with secrets stored in JSON format to create secrets from.
+    #[clap(long, name = "from-file", conflicts_with = "name")]
+    pub from_file: Option<PathBuf>,
+
+    /* --- Common args --- */
+    #[clap(flatten)]
+    #[allow(missing_docs)]
+    pub api: ApiOpts,
+
+    #[clap(flatten)]
+    pub env: WasmerEnv,
+
+    /// Don't print any message.
+    #[clap(long)]
+    pub quiet: bool,
+
+    /// Do not prompt for user input.
+    #[clap(long, default_value_t = !std::io::stdin().is_terminal())]
+    pub non_interactive: bool,
+}
+
+impl CmdAppSecretsCreate {
+    fn get_secret_name(&self) -> anyhow::Result<String> {
+        if let Some(name) = &self.secret_name {
+            return Ok(name.clone());
+        }
+
+        if self.non_interactive {
+            anyhow::bail!("No secret name given. Provide one as a positional argument.")
+        } else {
+            let theme = ColorfulTheme::default();
+            Ok(dialoguer::Input::with_theme(&theme)
+                .with_prompt("Enter the name of the secret")
+                .interact_text()?)
+        }
+    }
+
+    fn get_secret_value(&self) -> anyhow::Result<String> {
+        if let Some(value) = &self.secret_value {
+            return Ok(value.clone());
+        }
+
+        if self.non_interactive {
+            anyhow::bail!("No secret value given. Provide one as a positional argument.")
+        } else {
+            let theme = ColorfulTheme::default();
+            Ok(dialoguer::Input::with_theme(&theme)
+                .with_prompt("Enter the value of the secret")
+                .interact_text()?)
+        }
+    }
+
+    async fn get_app_id(&self, client: &WasmerClient) -> anyhow::Result<String> {
+        if let Some(app_id) = &self.app_id.app {
```

Comment:
Should re-use the logic from `AppIdent(Opts)`

Reply:
I chose not to use a function from `AppIdent*Flag*` as that part of the code was pretty tied into the logic of secrets and searching to fallback to user input. What I mean is that a function `AppIdentFlag::resolve() -> Result<AppIdent>` does not suit this piece of code, and `AppIdentFlag::search_ident(maybe_search_path) -> Result<Option<AppIdent>>` does not suit the other either. So, a utility function - that could, perhaps, use more methods from `app::utils` encloses the behaviour shared between the `secrets` commands to get an app identifier.